lib/kafka/client.rb

```diff
@@ -414,6 +414,32 @@ def last_offset_for(topic, partition)
       @cluster.resolve_offset(topic, partition, :latest) - 1
     end
 
+
+    # Retrieve the offset of the last message in each partition of the specfied topics.
+    #
+    # @param topics [String, Array<String>] single topic name or array of topic names.
+    #   nil means all topics will be fetched.
+    #
+    # @return [Hash]
+    #
+    # @example
+    #   last_offsets_for ['topic-1', 'topic-2']  ->
+    #   {
+    #     'topic-1' => { 0 => 100, 1 => 100 },
+    #     'topic-2' => { 0 => 100, 1 => 100 },
+    #   }
+    def last_offsets_for(topics = nil)
+      topics   = [topics] if !topics.nil? && topics.is_a?(String)
+      topics ||= self.topics
```

Comment:
I think this can be simplified a bit:

1) is there a reason to support fetching _all_ offsets? if not, let's not allow that for now.
2) you can use the splat operator to accept multiple arguments: `def last_offsets_for(*topics)`